tests/test_middleware.py

```diff
@@ -249,6 +266,23 @@ def allow_all(sender, request, **kwargs):
 
             assert allow_all.calls == 1
 
+    @override_settings(CORS_ORIGIN_WHITELIST=['example.com'])
+    @prepend_middleware('tests.test_middleware.ShortCircuitMiddleware')
+    def test_process_response_with_short_circuit(self):
+        """
+        Test a scenario when a middleware that returns a response is run before
+        the ``CorsMiddleware``. In this case
+        ``CorsMiddleware.process_response()`` should ignore the request if
+        MIDDLEWARE setting is used (new mechanism in Django 1.10+) and process
+        the request and add CORS response headers if MIDDLEWARE_CLASSES is
+        used in a backward compatible fashion with django-cors-headers pre 1.3.0.
+        """
+        resp = self.client.get('/', HTTP_ORIGIN='http://example.com')
+        if getattr(settings, 'MIDDLEWARE', None) is None:
```

Comment:
I'd gate on Django version rather than testing the settings, which are themselves just gated on Django version